examples/mini_repl.rs

```diff
@@ -0,0 +1,54 @@
+///! This example show cases a very simple REPL.
+///! While a much better REPL can be found in ../src/shell,
+///! This much smaller REPL is still a useful example because it showcases inserting
+///! values and functions into the Python runtime's scope, and showcases use
+///! of the compilation mode "Single".
+///! Note that in particular this REPL does a horrible job of showing users their errors
+///! (instead it simply crashes).
+use rustpython_compiler::compile;
+use rustpython_vm::{pyobject::PyResult, PySettings, VirtualMachine};
+// this needs to be in scope in order to insert things into scope.globals
+use rustpython_vm::pyobject::ItemProtocol;
+
+fn main() -> PyResult<()> {
+    let mut on = true;
```

Comment:
You can make this a `Rc<Cell<bool>>` to avoid the unsafe code later on; you could also put a comment here about why things like `Rc` and `Cell` and `RefCell` are necessary when interacting with RustPython (b/c Python doesn't respect Rust's borrow checker).